webapp/apps/btax/views.py

```diff
@@ -0,0 +1,393 @@
+    if model.tax_result:
+        output = url.unique_inputs.tax_result
+        first_year = url.unique_inputs.first_year
+        created_on = url.unique_inputs.creation_date
+        tables = btax_results_to_tables(output, first_year)
+        tables["tooltips"] = {
+            "TODO": "if tooltips needed, put them here"
+        }
+        inputs = url.unique_inputs
+        is_registered = True if request.user.is_authenticated() else False
+        context = tables.copy()
+        context.update({
+            'locals':locals(),
+            'unique_url':url,
+            'btax_version':BTAX_VERSION,
+            'taxcalc_version': TAXCALC_VERSION,
+            'tables': json.dumps(tables),
+            'created_on': created_on,
+            'first_year': first_year,
+            'is_registered': is_registered,
+            'is_micro': True
+        })
+        return render(request, 'btax/results.html', context)
+
+    else:
+
+        job_ids = model.job_ids
+        jobs_to_check = model.jobs_not_ready
+        if not jobs_to_check:
+            jobs_to_check = normalize(job_ids)
+        else:
+            jobs_to_check = normalize(jobs_to_check)
+
+        try:
+            jobs_ready = dropq_compute.dropq_results_ready(jobs_to_check)
+        except JobFailError as jfe:
+            print jfe
+            return render_to_response('taxbrain/failed.html')
+
+        if all(jobs_ready):
+            model.tax_result = dropq_compute.dropq_get_results(normalize(job_ids))
+
+            model.creation_date = datetime.datetime.now()
+            print 'ready'
+            model.save()
+            return redirect(url)
+        else:
+            jobs_not_ready = [sub_id for (sub_id, job_ready) in
+                                zip(jobs_to_check, jobs_ready) if not job_ready]
+            jobs_not_ready = denormalize(jobs_not_ready)
+            model.jobs_not_ready = jobs_not_ready
+            print 'not ready', jobs_not_ready
+            model.save()
+            if request.method == 'POST':
+                # if not ready yet, insert number of minutes remaining
+                exp_comp_dt = url.exp_comp_datetime
+                utc_now = datetime.datetime.utcnow()
+                utc_now = utc_now.replace(tzinfo=pytz.utc)
+                dt = exp_comp_dt - utc_now
+                exp_num_minutes = dt.total_seconds() / 60.
+                exp_num_minutes = round(exp_num_minutes, 2)
+                exp_num_minutes = exp_num_minutes if exp_num_minutes > 0 else 0
+                if exp_num_minutes > 0:
+                    return JsonResponse({'eta': exp_num_minutes}, status=202)
+                else:
+                    return JsonResponse({'eta': exp_num_minutes}, status=200)
+
+            else:
+                print "rendering not ready yet"
+                return render_to_response('btax/not_ready.html', {'eta': '100'}, context_instance=RequestContext(request))
+
+
+@permission_required('btax.view_inputs')
```

Comment:
we don't need the `csv_*` views, nor the `pdf*` view since that is handled at the JS layer now.